### DISCUSSION TITLE
Developper Setup PyOpenGL_Accelerate Python.h not found

### DISCUSSION THREAD

**Discussion by panthavma** [bnpr/Malt#279] [Q&A] [answered]
Hello! During developer setup, I wasn't able to complete the PyOpenGL_Accelerate part of install_dependancies.py, the error being that pip can't find Python.h during the compilation of the module.
Malt still seems to be able to run just fine for the moment, so this might not be critical.

Here's what I tried:
- Running the command explained in the documentation, fails when running pip install `PyOpenGL_Accelerate`
- Running `pip install PyOpenGL_Accelerate` in the console, it works.
- Running `pip install PyOpenGL_Accelerate --target <Malt Folder>/Malt/.Dependencies-310/`, it works
- Running `<Blender Python> -m pip install PyOpenGL_Accelerate --target <Malt Folder>/Malt/.Dependencies-310`, same error
- Copying the includes from the python source code to a include/python3.10 subfolder as per [https://blender.stackexchange.com/questions/81740/python-h-missing-in-blender-python](https://blender.stackexchange.com/questions/81740/python-h-missing-in-blender-python). Didn't work
- Install python3.10-dev from a custom ppa in order to have the include files on my machine for the correct version. Didn't work either.

Since the packages are installed on my machine and that I can build it, I suppose it's a problem with some kind of environment accessible from blender, although my knowledge is too limited to know how to search for it at the moment. Do you have any ideas of how to proceed ?

Here is my log if it helps.
```
Installing collected packages: PyOpenGL_Accelerate
  Running setup.py install for PyOpenGL_Accelerate ... error
  error: subprocess-exited-with-error
  
  × Running setup.py install for PyOpenGL_Accelerate did not run successfully.
  │ exit code: 1
  ╰─> [26 lines of output]
      running install
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-3.10
      creating build/lib.linux-x86_64-3.10/OpenGL_accelerate
      copying OpenGL_accelerate/__init__.py -> build/lib.linux-x86_64-3.10/OpenGL_accelerate
      running build_ext
      skipping 'src/wrapper.c' Cython extension (up-to-date)
      skipping 'src/formathandler.c' Cython extension (up-to-date)
      skipping 'src/arraydatatype.c' Cython extension (up-to-date)
      skipping 'src/errorchecker.c' Cython extension (up-to-date)
      skipping 'src/vbo.c' Cython extension (up-to-date)
      skipping 'src/nones_formathandler.c' Cython extension (up-to-date)
      skipping 'src/latebind.c' Cython extension (up-to-date)
      skipping 'src/buffers_formathandler.c' Cython extension (up-to-date)
      skipping 'src/numpy_formathandler.c' Cython extension (up-to-date)
      building 'OpenGL_accelerate.wrapper' extension
      creating build/temp.linux-x86_64-3.10
      creating build/temp.linux-x86_64-3.10/src
      gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -I/home/sybren/buildbot-builder/linux_glibc217_x86_64_cmake/build_deps/deps/Release/sqlite/include -I/home/sybren/buildbot-builder/linux_glibc217_x86_64_cmake/build_deps/deps/Release/bzip2/include -I/home/sybren/buildbot-builder/linux_glibc217_x86_64_cmake/build_deps/deps/Release/lzma/include -I/home/sybren/buildbot-builder/linux_glibc217_x86_64_cmake/build_deps/deps/Release/zlib/include -fPIC -fPIC -fPIC -I/tmp/pip-install-cfv0gko6/pyopengl-accelerate_bcfa7b7caffb4235b37c5bdb83360e29/.. -I/tmp/pip-install-cfv0gko6/pyopengl-accelerate_bcfa7b7caffb4235b37c5bdb83360e29/src -I/tmp/pip-install-cfv0gko6/pyopengl-accelerate_bcfa7b7caffb4235b37c5bdb83360e29 -I/home/pana/.software/BlenderMalt/blender-3.1.0-linux-x64/3.1/python/include/python3.10 -c src/wrapper.c -o build/temp.linux-x86_64-3.10/src/wrapper.o
      src/wrapper.c:4:10: fatal error: Python.h: No such file or directory
          4 | #include "Python.h"
            |          ^~~~~~~~~~
      compilation terminated.
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]

```

**Comment by pragma37** [comment, marked as answer]
PyOpenGL_Accelerate is fully optional.
It just may provide a slight performance boost to PyOpenGL, so I wouldn't worry too much if you can't get it to work.

What I do (on Windows) is to install the Python version (with the development headers) that matches the one used by Blender, and use that python installation instead of the one bundled by Blender to run the install scripts.

An easier option would be to just copy-paste the dependencies folder from the BlenderMalt releases from Github.

It may be desirable to add an option to the install scripts to simply fetch the dependencies and C libraries from a compiled release, since setting up the whole compiler toolchain can be difficult and most people won't need it anyway.

**Reply by panthavma** [reply]
Using the python executable directly seems to work, even with python 3.8. Thanks for the quick answer, I thought it was more complex than that :)

**Reply by pragma37** [reply]
> even with python 3.8

The compilation may have succeeded, but Malt won't load those dependencies unless the Python version in Blender matches.
Compiled Python modules are not compatible across versions.
